lib/splitclient-rb/engine/matchers/matcher.rb

```diff
@@ -30,5 +30,23 @@ def equals?(obj)
     def string_type?
       false
     end
+
+    private
+
+    def verify_semver_arg?(args, matcher_name)
+      @logger.debug("[#{matcher_name}] evaluating value and attributes.")
+      return false unless @validator.valid_matcher_arguments(args)
+
+      true
+    end
+
+    def check_semver_value_to_match(value_to_match, matcher_spec_name)
+      unless !value_to_match.nil? && !@semver.nil?
```

Comment:
Why denied? `if value_to_match.nil? || @semver.nil?`
If at least one is nil it simply returns false.